### PR TITLE
Graph scaladoc and tests

### PR DESCRIPTION
**Pull request by bogdanromanx**
Also bumped dependencies.

### CODE REVIEW

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
 
+import cats.Eq
 import cats.implicits._
 import ch.epfl.bluebrain.nexus.rdf.Graph._
 import ch.epfl.bluebrain.nexus.rdf.Iri.Path.Segment
 import ch.epfl.bluebrain.nexus.rdf.Iri.{AbsoluteIri, Url}
 import ch.epfl.bluebrain.nexus.rdf.Node.{BNode, IriNode, IriOrBNode}
 
+/**
+  * An RDF Graph anchored to a specific node. All vertices except the anchor are existential. Operations like append
```
Comment (wwajerowicz) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: What does `existential` mean in this context?

Comment (bogdanromanx) [reply modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: The vertices exist through the edge definitions only, cannot be asserted independently.

Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: A graph with a root node is a `RootedGraph`: https://en.wikipedia.org/wiki/Rooted_graph

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
+  * or prepend use the anchor to generate a relationship as a triple between the graph and the node / graph that gets
+  * appended or prepended.
+  */
 sealed abstract class Graph extends Product with Serializable {
 
+  /**
+    * The graph anchor. It represents the "main" node of the graph such that traversals have a default starting point.
```
Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: The root node, or starting point for traversals

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+    */
   def node: Node
```
Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: We could call this root instead to clarify, but node seems fine to me as well.

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 4]
```diff
@@ -36,61 +49,117 @@ sealed abstract class Graph extends Product with Serializable {
...
-  def append(g: Graph, predicate: IriNode): Graph =
+  /**
+    * Merges this graph with the provided one and additionally generates a new triple (this.node, predicate, g.node)
+    * that represents a relationship between the two subgraphs. The anchor of the resulting graph is `this.node`.
+    */
+  def append(predicate: IriNode, g: Graph): Graph =
     g.prepend(this, predicate)
 
+  /**
+    * Adds a new triple (this.node, p, o) to this graph yielding a new one. The anchor is retained.
```
Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: The root is retained

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 5]
```diff
@@ -36,61 +49,117 @@ sealed abstract class Graph extends Product with Serializable {
...
     Graph(node, triples + triple)
 
+  /**
+    * Adds the provided triples to the graph yielding a new one.
+    */
   def ++(triples: Set[Triple]): Graph =
     Graph(node, this.triples ++ triples)
 
+  /**
+    * Merges the this graph with that graph while retaining the anchor.
```
Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: retaining the root.

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 6]
```diff
@@ -151,7 +217,7 @@ sealed abstract class Graph extends Product with Serializable {
   private val dotNumeralRegex = "[-]?(.[0-9]+|[0-9]+(.[0-9]*)?)".r
 
   /**
-    * Returns DOT representation of [[Graph]].
+    * Returns a DOT representation of this graph.
```
Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: A DOT representation of this graph.

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala [hunk 7]
```diff
@@ -250,6 +316,9 @@ sealed abstract class Graph extends Product with Serializable {
 
 object Graph {
 
+  /**
+    * Constructs a new [[Graph]] from the provided anchor node and set of triples.
```
Comment (umbreak) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/Graph.scala]: from the provided root node

**Pull request merged**